kube-runtime/src/utils/predicate.rs

```diff
@@ -114,6 +115,33 @@ where
     }
 }
 
+/// Configuration for predicate filtering with cache TTL
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// Time-to-live for cache entries
+    ///
+    /// Entries older than this duration will be evicted from the cache.
+    /// Default is 1 hour.
+    pub ttl: Duration,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            // Default to 1 hour TTL - long enough to avoid unnecessary reconciles
+            // but short enough to prevent unbounded memory growth
```

Comment:
I am a bit uncertain about this value still. Because we bump the interval every time we see an object, **if we see the object again during relists** every 5m, then this setup (and value) works great because we will in practice never see the same object again unless the hash/uid changes.

However, I am noticing some differences in behaviour with long watches with and without using `.streaming_lists()`, and would like to confirm whether the behavior is correct there first.